### ISSUE TITLE
Strings should be unique per context, currently it is globally unique, different translation for same text doesn't work

### ISSUE THREAD

**Issue by n-for-all** [polylang/polylang#1063]
https://github.com/polylang/polylang/blob/f76147257848584b42822c14e3d7f3cba2707dc1/admin/admin-strings.php#L54

String can be unique per context as below
self::$strings[ md5( $string.$context ) ] = compact( 'name', 'string', 'context', 'multiline' );

**Comment by Chouby** [comment]
It's not as easy as doing just that. And there is probably a way around. Register 2 different strings. The content doesn't matter. So you can add some context directly in the source strings. Then translate the 2 source strings in all the languages, including the default one. This allows to display the same text for your 2 string in one language and different texts in other languages.

**Comment by n-for-all** [comment]
yes, that's a workaround, but the issue is that 2 plugins with the same translation name, overrides the other, modifying other authors plugin will break the updates.

It should work just like in wordpress, the domain in your plugin doesn't have any value whether you pass it or not, the translation is not per the plugin but global.

if possible, can you add a filter to file the key, i will register the domain and pass it through and the string to be unique per domain, so if the context is not Polylang add the context to the string key?
